### PR TITLE
feat: add protocol upgrade proposal api

### LINKED ISSUE #6552: Add API for protocol upgrade proposal information
In order to support the below AC we need to implement API(s?) to to provide information on the upcoming confirmed (protocol upgrade) proposal including total proposals/block details/versions.

([0075-PLUP-028](https://github.com/vegaprotocol/specs/blob/master/protocol/0075-PLUP_protocol_upgrades.md#0075-PLUP-028)) An datanode API should be available to provide information on the upcoming confirmed proposal including total proposals/block details/versions.

### PR DESCRIPTION
**Pull request by pscott31**
closes #6552

### CODE REVIEW

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -4568,3 +4571,40 @@ input DateRange {
 type ProtocolUpgradeStatus {
   ready: Boolean!
 }
+
+enum ProtocolUpgradeProposalStatus {
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:4575]: Can you put a description on this enum category?

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -4568,3 +4571,40 @@ input DateRange {
...
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_UNSPECIFIED
+  "Proposal to upgrade protocol version is awaiting sufficient validator approval"
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_PENDING
+  "Proposal to upgrade protocol version accepted"
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_APPROVED
+  "Proposal to upgrade protocol version has been rejected"
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_REJECTED
+}
+
+type ProtocolUpgradeProposalConnection {
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:4586]: this one too :)

File: datanode/gateway/graphql/schema.graphql [hunk 3]
```diff
@@ -4568,3 +4571,40 @@ input DateRange {
...
+
+"Edge type containing the protocol upgrade protocol cursor information"
+type ProtocolUpgradeProposalEdge {
+  "The protocol upgrade proposal"
+  node: ProtocolUpgradeProposal!
+  "Cursor identifying the protocol upgrade proposal"
+  cursor: String!
+}
+
+type ProtocolUpgradeProposal {
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:4601]: you know what I'm going to say

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -2059,8 +2062,25 @@ message GetProtocolUpgradeStatusResponse {
     bool ready = 1;
 }
 
+message ListProtocolUpgradeProposalsRequest {
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2065]: Could you also document all of the new ones here too?

**Review by dexturr — DISMISSED** [review]

File: datanode/gateway/graphql/schema.graphql (hunk 1 above)
Comment (pscott31) [reply datanode/gateway/graphql/schema.graphql:4575]: Yep sure

File: datanode/gateway/graphql/schema.graphql (hunk 2 above)
Comment (pscott31) [reply datanode/gateway/graphql/schema.graphql:4586]: yep yep

File: datanode/gateway/graphql/schema.graphql (hunk 3 above)
Comment (pscott31) [reply datanode/gateway/graphql/schema.graphql:4601]: :) all done

**Review by dexturr — DISMISSED** [review]

**Review by candida-d — DISMISSED** [review]

File: datanode/gateway/graphql/schema.graphql [hunk 5]
```diff
@@ -4568,3 +4571,43 @@ input DateRange {
...
+
+"Edge type containing the protocol upgrade protocol cursor information"
+type ProtocolUpgradeProposalEdge {
+  "The protocol upgrade proposal"
+  node: ProtocolUpgradeProposal!
+  "Cursor identifying the protocol upgrade proposal"
+  cursor: String!
+}
+
+"An proposal to upgrade the vega protocol (i.e. which version of the vega software nodes will run)"
```
[the remark this answers is not in the document]
Comment (pscott31) [reply datanode/gateway/graphql/schema.graphql]: Fixed thanks!

**Review by dexturr — APPROVED** [review]

**Review by candida-d — APPROVED** [review]

**Review by guoguojin — APPROVED** [review]

**Pull request merged**